base/base/FnTraits.h

```diff
@@ -16,8 +14,7 @@ struct FnTraits<R(A...)>
     static constexpr bool value = std::is_invocable_r_v<R, F, A...>;
 
     using Ret = R;
-    using Args = std::tuple<A...>;
-    using DecayedArgs = std::tuple<typename std::decay<A>::type...>;
+    using Args = TL<A...>;
```

Comment:
Motivation for that: FnTraits is used more often than CachedFn, and <tuple> header is quite heavy, so we can get rid of it here and move to CachedFn.